packages/core/src/lib/pagination/use-pagination.ts

```diff
@@ -16,38 +94,107 @@ export function usePagination<T>({
   /** The index of the active item. */
   active: number;
   /** Renders an item as part of a page. */
-  renderItem: (layout: Prettify<Layout<T>>) => string;
+  renderItem: (
+    layout: Prettify<{
+      item: T;
+      index: number;
+      isActive: boolean;
+    }>,
+  ) => string;
   /** The size of the page. */
   pageSize: number;
   /** Allows creating an infinitely looping list. `true` if unspecified. */
   loop?: boolean;
   theme?: Theme;
 }): string {
-  const state = useRef({ position: 0, lastActive: 0 });
+  const width = readlineWidth();
+  const state = useRef<{ lastPointer: number; lastActive: number | undefined }>({
+    lastPointer: active,
+    lastActive: undefined,
+  });
```

Comment:
I think we want `finite`/`infinite` to become `usePointerPosition()`; the ref isn't used anywhere else.